Release/0.0.1

Warps73 opened the pull request.

Commit:
Added circleci unit tests

Commit:
added php 5.6 version && test

Commit:
added php 5.6 version && test

Commit:
Updated Json file

Commit:
Updated Readme

Warps73 committed:
Completed Readme

Warps73 committed:
Merge branch 'feature/circle-ci' into develop

bouiboui commented on README.md:
```diff
@@ -1,29 +1,105 @@
 # MyElefant
 
-MyElefant is a service for send sms / push notification ect...
+MyElefant is a service for send sms / push notification etc...
```
```suggestion
MyElefant is a service for sending SMS messages / push notifications etc...
```

bouiboui commented on README.md:
```diff
@@ -1,29 +1,105 @@
 https://myelefant.com/
 
-This plugin is for send campaign with MyElefant's APIs.
+This plugin is for send sms campaign with MyElefant's APIs.
```
```suggestion
This plugin is for sending SMS campaigns with MyElefant's APIs.
```

bouiboui commented on README.md:
```diff
@@ -1,29 +1,105 @@
+
+**Default usage**
+
+For send a new campaign you have to provide many parameters : 
```
```suggestion
To send a new campaign you have to provide the following parameters: 
```

bouiboui commented on README.md:
```diff
@@ -1,29 +1,105 @@
+    Sender must be a string
+
+
+**For Message and Sender, you have 3 usages possible**
```
```suggestion
**For Message and Sender, you have 3 possible usages**
```

bouiboui commented on README.md:
```diff
@@ -1,29 +1,105 @@
+
+# Debug
 
+For activate log system change this parameter in MyElefant.yaml
```
```suggestion
To activate the logging system, change this parameter in MyElefant.yaml
```

bouiboui commented on composer.json:
```diff
@@ -1,18 +1,18 @@
-            "name": "Timothy",
-            "email": "Timothy.prefol@digitgroup.com"
+            "name": "PREFOL Timothy",
+            "email": "Timothy.prefol@digitregroup.com"
```
```suggestion
            "email": "timothy.prefol@digitregroup.com"
```

bouiboui commented on src/MyElefant.php:
```diff
@@ -1,5 +1,13 @@
 <?php
+/**
+ * Class MyElefant
+ * For send sms with myElefant service
```
```suggestion
 * To send SMS messages with myElefant service
```

bouiboui commented on src/MyElefant.php:
```diff
@@ -43,23 +47,26 @@ class MyElefant
 
     public function __construct(){
         $this->yamlDatas = Yaml::parseFile(__DIR__.'/../MyElefant.yaml');
```
`data` is already plural, `datas` is wrong.

bouiboui commented on src/MyElefant.php:
```diff
@@ -43,23 +47,26 @@ class MyElefant
-        $this->token = $this->getAuthentification(getenv('SECRET_KEY'));
-
-        if ($this->yamlDatas['APP_ENV'] == 'dev' ) {
+        if ($this->yamlDatas['APP_ENV'] == 'dev') {
```
Always use strict equality checks `===`

bouiboui commented on src/MyElefant.php:
```diff
@@ -43,23 +47,26 @@ class MyElefant
      * Initialize Dotenv
      * @return array
      */
     public function initDotEnv(){
```
If it's only called in this class, it should be `private`.
A good rule of thumb is to create classes as `private`, then adjust visibility to `protected` or `public` only when it makes sense.

bouiboui commented on src/MyElefant.php:
```diff
@@ -150,7 +157,7 @@ public function sendSms(array $contacts, $sendDate=null, $message = null, $sende
             $response = $client->request('POST', $this->yamlDatas['URL_MYELEFANT_API'].$this->yamlDatas['URL_MYELEFANT_API_CREATE_CAMPAIGN'],
                 [ 
-                'headers' =>['Authorization' => $this->token, 'Content-Type'=>'application/json'],
+                'headers' =>['Authorization' =>$this->getAuthentification(getenv('SECRET_KEY')), 'Content-Type'=>'application/json'],
```
It's a good practice to call `getenv` as early as possible in the code, because it makes it easier to debug and to test.
Even better if you can decouple it completely from the class, by requiring its value in the constructor.

bouiboui commented on src/tests/units/FakeMyElefant.php:
```diff
@@ -14,7 +14,7 @@ public function getAuthentification($secretKey){
 
-    public function sendSms(string $secretToken, array $contacts, $sendDate=null, $message = null, $sender = null){
+    public function sendSms($secretToken, $contacts, $sendDate=null, $message = null, $sender = null){
         if ($secretToken === 'validToken') {
```
Is this a mistake? `'validToken'`?

BasileTrujillo replied:
Here is the FakeMyElefant.php so that's fine, I think :)

Warps73 committed:
Updated library

-Removed .env file & package
-Removed yaml file & package
-Added config in abstract class
-Updated testing files
-Updated Readme

Warps73 committed:
Updated Library

Warps73 committed:
Updated composer packages

Warps73 committed:
Updated Composer packages

Warps73 committed:
test

BasileTrujillo commented on src/MyElefant.php:
```diff
@@ -1,239 +1,296 @@
+        if (isset($config['debug']) === true) {
+            $this->error = $this->initLogger(
+                'error',
+                'var/log/error.log'
```
can `var/log/error.log` be configurable through the $config object?

BasileTrujillo replied:
Same for `error`

Warps73 replied:
Yes, I will do it.

BasileTrujillo commented on src/MyElefant.php:
```diff
@@ -1,239 +1,296 @@
+            );
+            $this->info  = $this->initLogger(
+                'info',
+                'var/log/info.log'
```
can `var/log/error.log` be configurable through the $config object?
same for `info`?

BasileTrujillo committed:
Added composer cache by php version in pipeline

BasileTrujillo committed:
Added verbosity to atoum and Explicit loader

Warps73 committed:
Updated config
